pages/studentPortal/quizcomponent/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from "react";
+
+type SkillSelectionProps = {
+  selections: string[];
+};
+
+const SkillSelection: React.FC<SkillSelectionProps> = ({ selections }) => {
+  const [selected, setSelected] = useState<string[]>([]);
+
+  const handleSelection = (selection: string) => {
+    if (selected.includes(selection)) {
+      setSelected(selected.filter((s) => s !== selection));
+    } else {
+      setSelected([...selected, selection]);
+    }
+  };
+
+  return (
+    <div className="flex flex-col w-full max-w-4xl mx-auto p-4">
+      <div className="w-full mb-4 text-center">
+        <div className="font-bold text-2xl text-black-600"></div>
+      </div>
+      {selections.map((selection, index) => (
+        <div
+          key={index}
+          className={`flex   items-start justify-start w-full px-4 py-2 my-2  cursor-pointer ${
+            selected.includes(selection)
+              ? "bg-violet-300 text-black-500 border-2 border-black-500 rounded-[14px]"
+              : "bg-white text-black-600 border-2 border-black-500 rounded-[14px]"
+          }`}
+          onClick={() => handleSelection(selection)}
+        >
+          Input Text Selection {index + 1}
+        </div>
+      ))}
+    </div>
+  );
+};
+
+type ProgressBarProps = {
+  progress: number;
+};
+
+const ProgressBar: React.FC<ProgressBarProps> = ({ progress }) => {
```

Comment:
could use of extracting components out into different functions, as the feature grows, it would be good to have each component live in it's own file all inside the same folder.

Reply:
Understood. Will organize my components before merging.